File: third-person/public/render_tile.js
class RenderTile {
	static Side(buffer, side, x, y, z, texture, rgb_a, rgb_b, rgb_c, rgb_d) {
		switch (side) {
			case WORLD_POSITIVE_X:
				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]
				break
			case WORLD_NEGATIVE_X:
				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]
				break
			case WORLD_POSITIVE_Y:
				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]
				break
			case WORLD_NEGATIVE_Y:
				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]
				break
			case WORLD_POSITIVE_Z:
				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z + 1.0
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]
				break
			case WORLD_NEGATIVE_Z:
				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_a[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_a[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[1]

				buffer.vertices[buffer.vertex_pos++] = x
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_b[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_b[2]
				buffer.vertices[buffer.vertex_pos++] = texture[0]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y + 1.0
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_c[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_c[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[3]

				buffer.vertices[buffer.vertex_pos++] = x + 1.0
				buffer.vertices[buffer.vertex_pos++] = y
				buffer.vertices[buffer.vertex_pos++] = z
				buffer.vertices[buffer.vertex_pos++] = rgb_d[0]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[1]
				buffer.vertices[buffer.vertex_pos++] = rgb_d[2]
				buffer.vertices[buffer.vertex_pos++] = texture[2]
				buffer.vertices[buffer.vertex_pos++] = texture[1]
				break
		}

		if (Render.Lumin(rgb_a) + Render.Lumin(rgb_c) < Render.Lumin(rgb_b) + Render.Lumin(rgb_d)) {
			Render.MirrorIndex4(buffer)
		} else {
			Render.Index4(buffer)
		}
	}
}

File: client/render/render-tile.go
package render

import "../graphics"

// RendTilePosX func
func RendTilePosX(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}

// RendTileNegX func
func RendTileNegX(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}

// RendTilePosY func
func RendTilePosY(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}

// RendTileNegY func
func RendTileNegY(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}

// RendTilePosZ func
func RendTilePosZ(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}

// RendTileNegZ func
func RendTileNegZ(b *graphics.RenderBuffer, x, y, z float32, texture *[4]float32, rgb *[4][3]float32) {
	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[0][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[1][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[1]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[2][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	b.Vertices[b.VertexPos] = x + 1.0
	b.VertexPos++
	b.Vertices[b.VertexPos] = y
	b.VertexPos++
	b.Vertices[b.VertexPos] = z
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][1]
	b.VertexPos++
	b.Vertices[b.VertexPos] = rgb[3][2]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[0]
	b.VertexPos++
	b.Vertices[b.VertexPos] = texture[3]
	b.VertexPos++

	if Lumin(&rgb[0])+Lumin(&rgb[2]) < Lumin(&rgb[1])+Lumin(&rgb[3]) {
		MirrorIndex4(b)
	} else {
		Index4(b)
	}
}
